docs/docs/aztec/smart_contracts/injecting_data/index.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Injecting Data
+tags: [functions, oracles, capsules]
+---
+
+There are multiple ways to inject data into smart contracts.
+
+1. **Oracles** - fetching data from the outside world
+2. **Authwits (Authentication Witnesses)** - authorizing an arbitrary action (or piece of data)
+3. **Capsules** - local data storage in the PXE
+
+## Oracles
+
+In the world of EVM, you can read data directly from storage. However, when we are working with circuits for private execution, this becomes more tricky as you cannot just read the storage directly from your state tree, because there are only commitments (e.g. hashes) there. The pre-images (content) of your commitments need to be provided to the function to prove that you actually allowed to modify them.
```

Comment:
```suggestion
In the world of EVM, you can read data directly from storage. However, when we are working with circuits for private execution, this becomes more tricky as you cannot just read the storage directly from your state tree, because there are only commitments (e.g. hashes) there. The pre-images (content) of your commitments need to be provided to the function to prove that you know value that the commitments represent and that you are actually allowed to modify them.
```